lib/util/ssr.js

```diff
@@ -6,13 +6,45 @@
  */
 'use strict';
 
+let ssrEscapeFound = false;
+
 module.exports.isSSREscape = function isSSREscape(node) {
-    return (
-        (node.type === 'IfStatement' || node.type === 'ConditionalExpression') &&
-        (isMetaEnvCheck(node.test) || isWindowOrDocumentCheck(node.test))
-    );
+    if (node.type === 'IfStatement' || node.type === 'ConditionalExpression') {
+        ssrEscapeFound = false;
+        if (checkConditionalStatements(node.test) || isWindowOrDocumentCheck(node.test)) {
```

Comment:
Since `typeof window !== 'undefined'` is basically the same as `!import.meta.env.SSR`, I think we should handle the recursive case here too:

```js
if (randomOtherCheck && typeof window !== 'undefined') {
```